contents/overlay/dialogs/gdpr-cookie-dialog/index.tsx

```diff
@@ -0,0 +1,75 @@
+import {
+  Button,
+  Dialog,
+  Link,
+  Tab,
+  TabPanel,
+  Tabs,
+  useDisclosure,
+} from "@yamada-ui/react"
+import { type FC, useState } from "react"
+import { CategoryItem } from "./category-item"
+import { CookieData } from "./data"
+
+const ComplexCookieDialog: FC = () => {
+  const { isOpen, onOpen, onClose } = useDisclosure()
+  const [categoryCheck, setCategoryCheck] = useState<{
+    [key: number]: boolean
+  }>({})
```

Comment:
Should be pass initial values as arguments?

```suggestion
CookieData.reduce((acc, item) => {
    acc[item.id] = !(item.isChecked);
    return acc;
  }, {} as Record<number, boolean>)
```

Reply:
I didn't see the need to pass initial values as arguments. Was it not sufficient to just set initial values in data.ts?
If possible, I would like to know why.

Reply:
If you want to set an initial value, you would want the following code.

```suggestion
 (cookieData.reduce((acc, item) => {
    acc[item.id] = item.isChecked ?? false;
    return acc;
  }, {} as Record<number, boolean>));
```

Reply:
> I didn't see the need to pass initial values as arguments. Was it not sufficient to just set initial values in data.ts? If possible, I would like to know why.

I thought it would be more natural for categoryCheck to have the values of all items when onSuccess is executed.

```tsx
        onSuccess={() => {
          onClose()
          console.log(categoryCheck); // { 0: true, 1: false: 2: false ... }
        }}
```

Reply:
Indeed, I think that is how it should be. Thank you very much for your guidance.